Drift/Private/DriftEventManager.cpp

```diff
@@ -57,27 +67,57 @@ TStatId FDriftEventManager::GetStatId() const
 
 void FDriftEventManager::FlushEvents()
 {
+    SCOPE_CYCLE_COUNTER(STAT_FlushDriftEvents);
+
     if (eventsUrl.IsEmpty())
     {
         return;
     }
 
     if (pendingEvents.Num() != 0)
     {
-        auto rm = requestManager.Pin();
-        if (rm.IsValid())
+        if (!requestManager.IsValid())
+        {
+            UE_LOG(LogDriftEvent, Error, TEXT("Failed to flush events. Request manager is invalid."));
+            return;
+        }
+        
+        UE_LOG(LogDriftEvent, Verbose, TEXT("[%s] Drift flushing %i events..."), *FDateTime::UtcNow().ToString(), pendingEvents.Num());
+
+        AsyncTask(ENamedThreads::AnyBackgroundThreadNormalTask, [this, Events = MoveTemp(pendingEvents)]()
```

Comment:
Capturing "this" is not safe and unnecessary, its only purpose here is to provide access to the requestManager so why not capture that directly?

Reply:
Because the callback may happen after the request manager has been reset internally, in both levels. The solution is to capture a weak pointer to the event manager instead of this. Inherit from TSharedFromThis, and capture WeakThis instead, like

`TWeakPtr<FSomeStruct> WeakThis = SharedThis(this)`